File: recursion/sumofNnumbers.cpp
#include<iostream>
using namespace std;
int sumofNnumbers(int N){
    int sum;
    if (N==0){
        return 0;
    }
    else {
        sum=N+sumofNnumbers(N-1);
    }
    return sum;
}
int main(){
    cout<<(sumofNnumbers(4));
}

File: recursion/sumofNnumbers.java
import java.util.Scanner;
public class sumofNnumbers {
    static int sumofNnumbers(int N){
        int sum;
        if(N==0){return 0;}
        else{return N+sumofNnumbers(N-1);}
    }
    public static void main(String[] args) {
        System.out.println(sumofNnumbers(5));

    }
}
